client/components/Overview.jsx

```diff
@@ -4,33 +4,62 @@ import ProductInformation from './overviewWidget/ProductInformation.jsx';
 import StyleSelector from './overviewWidget/StyleSelector.jsx';
 import AddToCart from './overviewWidget/AddToCart.jsx';
 import ProductOverview from './overviewWidget/ProductOverview.jsx';
-// import axios from 'axios';
+import axios from 'axios';
 
 class Overview extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
+      styles: [],
       selectedStyle: null
     };
+
+    this.clickedStyleHandler = this.clickedStyleHandler.bind(this);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.product !== prevProps.product) {
```

Comment:
If you wanted I think this could be even more targeted since you only use ```this.props.product.product.id``` so instead of comparing the whole product just compare the id.